crates/re_viewer_context/src/space_view/space_view_class.rs

```diff
@@ -83,19 +64,41 @@ pub trait SpaceViewClass: std::marker::Sized {
         space_view_id: SpaceViewId,
     );
 
+    /// Executed before the ui method is called, can be use for heuristic & state updates before populating the scene.
+    ///
+    /// Is only allowed to access archetypes defined by [`Self::blueprint_archetype`]
+    /// Passed entity properties are individual properties without propagated values.
+    fn prepare_ui(
+        &self,
+        _ctx: &mut ViewerContext<'_>,
+        _state: &Self::State,
+        _entity_paths: &IntSet<EntityPath>,
+        _entity_properties: &mut re_data_store::EntityPropertyMap,
+    ) {
+    }
+
     /// Draws the ui for this space view class and handles ui events.
     ///
-    /// The passed scene is already populated for this frame
     /// The passed state is kept frame-to-frame.
+    ///
+    /// The passed systems (`view_ctx` and `parts`) are only valid for the duration of this frame and
+    /// were already executed upon entering this method.
+    ///
+    /// `draw_data` is all draw data gathered by executing the view part systems.
+    /// TODO(wumpf): Right now the ui methods control when and how to create [`re_renderer::ViewBuilder`]s.
+    ///              In the future, we likely want to move view builder handling to `re_viewport` with
+    ///              minimal configuration options exposed via [`crate::SpaceViewClass`].
+    #[allow(clippy::too_many_arguments)]
```

Comment:
having so many arguments here pains me. `draw_data` might go away in the future if we enforce a (single..?) re_renderer view builder for each space view. But otherwise I'm stumped, all of them seem to be important.
Could consider putting all systems in a struct, but that's a weird micro-optimization